Sources/StreamChat/Controllers/ConnectionController/ConnectionController.swift

```diff
@@ -92,11 +92,15 @@ public extension ChatConnectionController {
             }
         }
     }
-
-    /// Disconnects the chat client the controller represents from the chat servers.
-    /// No further updates from the servers are received.
-    func disconnect() {
-        chatClientUpdater.disconnect()
+    
+    /// Stops active web-socket connection for the underlaying `ChatClient` and
+    /// cancels all queued API requests. If there is a connection recovery flow running, it will be interrupted.
+    ///
+    /// - Parameter completion: The completion handler invoked when the `ChatClient` has been disconnected.
+    func disconnect(completion: (() -> Void)? = nil) {
+        chatClientUpdater.disconnect {
```

Comment:
@polqf passing `completion` to `chatClientUpdater.disconnect` is not possible since the closure is optional here but updater requires a non-optional one.

What we could is:
```swift
chatClientUpdater.disconnect(completion: completion ?? {})
```

Reply:
Sonar cloud treats `completion ?? {}` as a code smell. Let's keep the current version